pkg/gateway/middleware/tenant.go

```diff
@@ -0,0 +1,41 @@
+package middleware
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/skygeario/skygear-server/pkg/gateway/model"
+	"github.com/skygeario/skygear-server/pkg/core/config"
+)
+
+type TenantMiddleware struct {
+}
+
+func (a TenantMiddleware) Handle(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		host := r.Host
+		app := model.GetApp(host)
```

Comment:
`model.GetApp(r)`?

Reply:
Will update it when updating query from db:)